modules/controller_pool/controller-primary.tpl

```diff
@@ -2,10 +2,37 @@
 
 export HOME=/root
 
+function load_workloads() {
+  mkdir /root/kube ; \
+  echo ${workloads} | sed 's| |\n|'g | awk '{sub(/:/," ")}1' | tee /root/workloads.data && \
```

Comment:
Why are we creating a text file with the json contents? And then reconstructing the json file?

We could provision the workloads json with a file provisioned, or perhaps better yet, userdata.

https://registry.terraform.io/providers/hashicorp/cloudinit/latest/docs/data-sources/cloudinit_config
Include any existing cloud-config, shell scripts, needed python scripts, and this json file as "parts".  And then `userdata = data.cloudinit_config.foo.rendered`

Reply:
The way I did this definitely needs some refining, but the original reason is that `$workloads` when passed as a `jsonencode()` string from the template, was not in JSON but `key:value key2:value key3:value`, so in my haste to get this part working (so I could reference them elsewhere down-script), I chose this path but at the very least, I don't need a file to read then creating another file, so that's at least one thing I can cut out. 

I was unaware of this data source, so I may revisit this now as well.